File: Algorithms/1437_Check_If_All_1s_Are_at_Least_Length_K_Places_Away/Solution.scala
object Solution {
  def kLengthApart(nums: Array[Int], k: Int): Boolean = {
    val arr = nums.zipWithIndex
      .filter(_._1 == 1)
      .map(_._2)
    arr.isEmpty || !arr.zip(arr.tail).exists(x => x._2 - x._1 - 1 < k)
  }
}

File: Algorithms/1437_Check_If_All_1s_Are_at_Least_Length_K_Places_Away/Solution.kt
class Solution {
    fun kLengthApart(nums: IntArray, k: Int): Boolean = nums.zip(nums.indices)
        .filter { it.first == 1 }
        .map { it.second }
        .run {
            this.isEmpty() || this.zipWithNext().find { it.second - it.first - 1 < k } == null
        }
}